### PR TITLE
[network] dont combine connection states

### PR DESCRIPTION
**Pull request by gmmeyer**
This supercedes #2856. It fixes the failing tests and puts the changes behind a flag.

### CODE REVIEW

File: conf.d/network.yaml.default [hunk 1]
```diff
@@ -13,3 +13,6 @@ instances:
     # Optionally completely ignore any network interface
     # matching the given regex:
     # excluded_interface_re: my-network-interface.*
+
+    # Do not combine connection states
```
Comment (olivielpeau) [inline conf.d/network.yaml.default]: Could you add more details on what the option does? And I think it's worth explicitly mentioning what the default value is.

Comment (gmmeyer) [inline conf.d/network.yaml.default]: That's a good note, thank you.

**Comment by gmmeyer** [comment]
@olivielpeau I can add some tests, however we'll have to add them in integrations core, the changes to the file structure forced me to change the tests for this one pretty substantially :(

**Comment by gmmeyer** [comment]
I'll put it in the issues there

**Pull request merged**